lib/vets/collections/pagination.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'common/exceptions/invalid_pagination_params'
+
+module Vets
+  module Collections
+    class Pagination
+      attr_reader :data
+
+      def initialize(page:, per_page:, total_entries:, data:, use_will_paginate: false)
+        @page = page
+        @per_page = per_page
+        @total_entries = total_entries
+
+        @data = if use_will_paginate && defined?(::WillPaginate::Collection)
+                  will_paginate_collection(data)
+                else
+                  data[((page - 1) * per_page)...(page * per_page)]
+                end
+      end
+
+      def metadata
+        {
+          pagination: {
+            current_page: @page,
+            per_page: @per_page,
+            total_pages: total_pages,
+            total_entries: @total_entries
+          }
+        }
+      end
+
+      private
+
+      def total_pages
+        (@total_entries / @per_page.to_f).ceil
```

Comment:
Would there be issue here if total entries is 0? Maybe in the case of 0 total entries, we could just return 1